From 20d2095d85582662722c43f2b368bba676e4a790 Mon Sep 17 00:00:00 2001
From: Christian Aistleitner <christian.aistleitner@gmx.at>
Date: Mon, 14 Jul 2008 06:56:00 -0300
Subject: [PATCH] Respect rotation upon ShadowFB initialization

fixes ShadowFB initialization while respecting rotation

The shadowFB is initialized using the screen's VirtualX for width and
VirtualY for height and regardless of the rotation.

When using the
  Option "Rotate" "CW"
such initialization of the ShadowFB results in such a setup

331
331
22

although the correct setup would be

33
33
33

the 1s and 3s represent the usable workspace.
the 2s and 3s represent the monitor.

Signed-off-by: Paulo Cesar Pereira de Andrade <pcpa@mandriva.com.br>
---
 src/s3v_driver.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/s3v_driver.c b/src/s3v_driver.c
index 499d2cf..0f605df 100644
--- xf86-video-s3virge-1.10.1/src/s3v_driver.c
+++ xf86-video-s3virge-1.10.1/src/s3v_driver.c
@@ -2651,8 +2651,8 @@ S3VInternalScreenInit( int scrnIndex, ScreenPtr pScreen)
 	case 16:
 	case 24:
 	case 32:
-	    ret = fbScreenInit(pScreen, FBStart, pScrn->virtualX,
-			       pScrn->virtualY, pScrn->xDpi, pScrn->yDpi,
+	    ret = fbScreenInit(pScreen, FBStart, width,
+			       height, pScrn->xDpi, pScrn->yDpi,
 			       displayWidth, pScrn->bitsPerPixel);
 	    break;
 	default:
-- 
1.5.6.5

